### ISSUE TITLE
Modify runner difficulty from BroncoSaveGame

### ISSUE THREAD

**Issue by timblegoorn** [cs481-ekh/f22-kitty-cat-studios#311]
References #308 

Set the runner difficulty based on the current BroncoSaveGame. If there is no save game (map loaded in editor for example), default to medium difficulty.

**Comment by timblegoorn** [comment]
Put in about an hour deliberating the best way to implement this (if it should go through the spawner controller or the runner or AIactor, etc.) ultimately, I decided the AISpawnerController will set the difficulty as it was built in to its functionality and this means there will only be one call to load the save game instead of a call every time a runner is spawned. It's not finished yet, still need to test, but I theoretically have it mostly done. Currently the AI will now spawn with a default difficulty and the spawner controller will overwrite that difficulty based on the save game difficulty.

**Comment by timblegoorn** [comment]
Put in an hour and a half to fully implement the difficulty settings. Health mod, damage mod, and fire rate mod were set but never applied. I applied the values to the actual runner parameters and tweaked the values to be more appropriate for each difficulty setting. In addition, I had to tweak the accuracy modification as the accuracy modifier was only being set once and not dynamically updated, so I created a new function to generate a unique shot modifier vector per shot based on the accuracy setting. Hard feels genuinely hard, medium is a good challenge, and easy feels like a great place to start. But, the modifiers may be tweaked in the future based on feedback.

Overall, approximately 2.25 hours spent on this task.
